libs/blocks/preflight/panels/performance.js

```diff
@@ -0,0 +1,277 @@
+});
+
+export const createPerformanceItem = ({
+  icon,
+  title,
+  description,
+} = {}) => html`<${PerformanceItem}
+  icon=${icon}
+  title=${title}
+  description=${description}
+/>`;
+
+let clonedLcpSection;
+function highlightElement(event) {
+  if (!config.lcp) return;
+  const lcpSection = config.lcp?.element.closest('.section');
+  const tooltip = document.querySelector('.lcp-tooltip-modal');
+  const { offsetHeight, offsetWidth } = lcpSection;
+  const scaleFactor = Math.min(500 / offsetWidth, 500 / offsetHeight);
+  if (!clonedLcpSection) {
+    clonedLcpSection = lcpSection.cloneNode(true);
+    clonedLcpSection.classList.add('lcp-clone');
+    clonedLcpSection.style.width = `${lcpSection.offsetWidth}px`;
+    clonedLcpSection.style.height = `${lcpSection.offsetHeight}px`;
+    clonedLcpSection.style.transform = `scale(${scaleFactor})`;
+    clonedLcpSection.style.transformOrigin = 'top left';
+  }
+  if (!tooltip.children.length) tooltip.appendChild(clonedLcpSection);
+  const { top, left } = event.currentTarget.getBoundingClientRect();
+  tooltip.style.width = `${offsetWidth * scaleFactor}px`;
+  tooltip.style.height = `${offsetHeight * scaleFactor}px`;
+  tooltip.style.top = `${top + window.scrollY - offsetHeight * scaleFactor - 10}px`;
+  tooltip.style.left = `${left + window.scrollX}px`;
+  document.querySelector('.lcp-tooltip-modal').classList.add('show');
+}
+
+const removeHighlight = () => { document.querySelector('.lcp-tooltip-modal').classList.remove('show'); };
+
+function observePerfMetrics() {
+  // Observe LCP
+  const lcpObserver = new PerformanceObserver((entryList) => {
+    const entries = entryList.getEntries();
+    const lastEntry = entries[entries.length - 1];
+    if (lastEntry) config.lcp = lastEntry;
+    const hasValidLcp = checkLCP();
+    if (!hasValidLcp) {
+      Object.values(text).forEach(({ key }) => {
+        if (key === 'lcpEl') return;
+        updateItem({ key, description: 'No LCP element found.' });
+      });
+      return;
+    }
+    checkFragments();
+    checkForPersonalization();
+    checkVideosWithoutPosterAttribute();
+    checkIcons();
+    checkForSingleBlock();
+    Promise.all([checkImageSize(), checkPlaceholders()]);
+  });
+  lcpObserver.observe({ type: 'largest-contentful-paint', buffered: true });
+
+  // Observe CLS
+  const clsObserver = new PerformanceObserver((entryList) => {
+    const entries = entryList.getEntries();
+    entries.forEach((entry) => {
+      if (!entry.hadRecentInput) {
+        config.cls += entry.value;
+      }
+    });
+    if (config.cls > 0) {
+      // TODO - Lana log? We should not have any CLS.
+      // console.log('CLS:', cls);
+    }
+  });
+  clsObserver.observe({ type: 'layout-shift', buffered: true });
+}
+
+export function Panel() {
+  useEffect(() => observePerfMetrics(), []);
+
+  return html`
+    <div class="seo-columns">
+      <div class="seo-column">${config.items.value.slice(0, 4).map((item) => createPerformanceItem(item))}</div>
```

Comment:
`seo-column` may need to be renamed, since its properties are now used for this panel as well.